agent/healthcheck/health_check.go

```diff
@@ -208,15 +222,37 @@ func (healthStatus *HealthStatus) computeHealthCheck(agentConfig config.AgentCon
 			healthStatus.HealthStatus = Healthy
 			break
 		}
-		if healthStatus.ManagementServerConnectionStatus == connected {
-			healthStatus.HealthStatus = Healthy
-		} else {
-			if time.Since(*healthStatus.ManagementServerDisconnectedTimestamp) < agentConfig.HcDisconnectedTimeout {
-				healthStatus.HealthStatus = Healthy
-			} else {
-				healthStatus.HealthStatus = Unhealthy
-			}
-		}
+
+    // If ManagementServerConnectionStatus has changed since the last update, add the current time to FlipTimestamps.
+        if healthStatus.ManagementServerConnectionStatus != healthStatus.LastConnectionStatus {
+            healthStatus.FlipTimestamps = append(healthStatus.FlipTimestamps, currentTime)
+
+            // Remove timestamps older than 30 minutes from FlipTimestamps.
+            thirtyMinutesAgo := currentTime.Add(-30 * time.Minute)
+            count := 0
+            for i, timestamp := range healthStatus.FlipTimestamps {
+                if timestamp.After(thirtyMinutesAgo) {
+                    break
+                }
+                count = i + 1
+            }
+            healthStatus.FlipTimestamps = healthStatus.FlipTimestamps[count:]
+
+            // Update LastConnectionStatus with the current ManagementServerConnectionStatus.
+            healthStatus.LastConnectionStatus = healthStatus.ManagementServerConnectionStatus
+        }
+
+        if healthStatus.ManagementServerConnectionStatus == connected {
+            healthStatus.HealthStatus = Healthy
+            healthStatus.FlipTimestamps = nil // clear the timestamps as the connection is now stable
```

Comment:
I don't think we should be doing this part, because if the connected status is "1" then we will clear the past history.

Reply:
Nice catch. Will remove this logic.